add Rakefile config.project to set changelog project name

jhoblitt opened the pull request:
For example, in `.sync.yml`:

    Rakefile:
      config.project: puppet-ipmi

smortex commented on moduleroot/Rakefile.erb:
```diff
@@ -52,7 +52,7 @@ begin
     config.exclude_labels = %w{duplicate question invalid wontfix wont-fix modulesync skip-changelog}
     config.user = '<%= @configs['config.user'] -%>'
-    config.project = metadata.metadata['name']
+    config.project = <%= @configs['config.project'] || "metadata.metadata['name']" %>
```
Some extra quoting is needed I think:
```suggestion
    config.project = <%= @configs['config.project']&.inspect || "metadata.metadata['name']" %>
```

ekohl replied:
I think it's safe to assume a project doesn't have `'` in it, so this would also work:
```suggestion
    config.project = '<%= @configs['config.project'] || metadata.metadata['name'] %>'
```

jhoblitt replied:
I like both of the proposed approaches.  I think @ekohl's approach might be slightly better in that in results in single quotes, which would be better for if/when we start allowing `rubocop` to lint the `Rakefile`.

jhoblitt commented:
:facepalm: `metadata.metadata['name']` needs to be a literal string...

jhoblitt commented:
3rd time is the charm...

jhoblitt commented:
Nope... because `!!''` is true.

jhoblitt commented:
I've manually tested the 4th attempt as working both when `config.project` is set and it is not.  This is probably a sign that we need a test harness in this repo...

ekohl commented on moduleroot/Rakefile.erb:
```diff
@@ -52,7 +52,7 @@ begin
     config.exclude_labels = %w{duplicate question invalid wontfix wont-fix modulesync skip-changelog}
     config.user = '<%= @configs['config.user'] -%>'
-    config.project = metadata.metadata['name']
+    config.project = <%= @configs['config.project'] ? "'#{@configs['config.project']}'" : "metadata.metadata['name']" %>
```
Modulesync also has this info. Twice actually:

```suggestion
    config.project = '<%= @metadata[:module_name] %>'
```

That's defined [here](https://github.com/voxpupuli/modulesync/blob/ac051ca961c1119a4b731bf990e8db0be6cf497b/lib/modulesync.rb#L95-L100)

It's also in `@configs` which would be more consistent with the previous line:
```suggestion
    config.project = '<%= @configs[:puppet_module] %>'
```

IMHO metadata is cleaner, but we should also add namespace to `metadata`: https://github.com/voxpupuli/modulesync/pull/254

jhoblitt replied:
I really like this. This should work for non-voxpupuli users without having to put anyting in the `.sync.yml`. :100:

jhoblitt replied:
I'm going to change this slightly so that it can be overridden, if need be.

ekohl commented on moduleroot/Rakefile.erb:
```diff
@@ -52,7 +52,7 @@ begin
     config.header = "# Changelog\n\nAll notable changes to this project will be documented in this file.\nEach new release typically also includes the latest modulesync defaults.\nThese should not affect the functionality of the module."
     config.exclude_labels = %w{duplicate question invalid wontfix wont-fix modulesync skip-changelog}
     config.user = '<%= @configs['config.user'] -%>'
```
I realized we can remove our hardcoding in `config_defaults.yml` for this:
```suggestion
    config.user = '<%= @configs['config.user'] || @configs[:namespace] %>'
```

ekohl reviewed (commented):
I realized this is another way to go: rely on native modulesync which already has the config.

jhoblitt commented:
@ekohl I've implemented your changes and removed `config.user` from `config_defaults.yml`. The diff from `--noop` looks good to me:

```Syncing 'puppet-allknowingdns'
Using no-op. Files in 'puppet-allknowingdns' may be changed but will not be committed.
Files changed:
diff --git a/Rakefile b/Rakefile
index f92f051..e5863a2 100644
--- a/Rakefile
+++ b/Rakefile
@@ -52,7 +52,7 @@ begin
     config.header = "# Changelog\n\nAll notable changes to this project will be documented in this file.\nEach new release typically also includes the latest modulesync defaults.\nThese should not affect the functionality of the module."
     config.exclude_labels = %w{duplicate question invalid wontfix wont-fix modulesync skip-changelog}
     config.user = 'voxpupuli'
-    config.project = metadata.metadata['name']
+    config.project = 'puppet-allknowingdns'
   end
 
   # Workaround for https://github.com/github-changelog-generator/github-changelog-generator/issues/715
Files added:


--------------------------------
```

ekohl commented:
It does want signed commits in this repo.
